pkg/arvo/lib/btc-node-json.hoon

```diff
@@ -0,0 +1,3129 @@
+/-  *btc-node-hook
+/+  base64
+=,  format
+=>  =,  dejs
+    |%
+    ::  %ferm: Checks the unit for ~ and returns a json e.g. [%s @ta]
+    ::
+    ++  ferm  |*([a=(unit) t=term] ?~(a ~ t^u.a))
+    ::  %feud: Checks the unit for ~ and returns a [%n @ud] json
+    ::
+    ++  feud  |=(a=(unit @u) ?~(a ~ (numb:enjs u.a)))
+    ::  %method: Removes 'hep' (-) from a %tas producing the RPC method
+    ::
+    ::  (e.g. %add-multisig-address -> 'addmultisigaddress')
+    ::
+    ++  method
+      |=  t=@t
+      ^-  @t
+      %+  scan  (scow %tas t)
+      ((slug |=(a=[@ @t] (cat 3 a))) (star hep) alp)
+    ::  %groups: used in %list-address-groupings
+    ::
+    ++  groups
+      |=  l=(list @t)
+      ^-  (list [?(@uc [%bech32 @t]) @t (unit @t)])
+      ?>  ?=([@t @t *] l)
+      :_  ~
+      :*  (addr-type-validator i.l)
+        ::
+          i.t.l
+        ::
+          ?~  t.t.l
+            ~
+          (some i.t.t.l)
+      ==
+    ::  %base58-to-cord: parses @uc to BTC addresses (legacy and p2sh)
+    ::
+    ++  base58-to-cord
+      |=  b=@uc
+      ^-  @t
+      ::  Removes leading 0c
+      ::
+      (rsh 3 2 (scot %uc b))
+    ::  %hex-to-cord: parses hexadecimal to cords without dots and 0x
+    ::
+    ++  hex-to-cord
+      |=  h=@ux
+      ^-  @t
+      %-  crip
+      =-  ((x-co:co (mul 2 p)) q)
+      (as-octs:mimes:html h)
+    ::  %addr-type-validator: parses BTC addresses (legacy and p2sh) to @uc
+    ::
+    ::    bech32/segwit addressed are tagged separately.
+    ::
+    ++  addr-type-validator
+      |=  addr=@t
+      ^-  ?(@uc [%bech32 @t])
+      =,  dejs
+      ?~  (rush addr fim:ag)
+        ::  TODO: fim:ag doesn't parse %bech32 addresses
+        ::
+        bech32+addr
+      `@uc`(rash addr fim:ag)
+    ::  %to-hex: parses hexadecimal to @ux with separator dots and 0x
+    ::
+    ++  to-hex
+      |=  h=@t
+      ^-  @ux
+      ?:  =('' h)  0x0
+      ::  Add leading 00
+      ::
+      =+  (lsh 3 2 h)
+      ::  Group by 4-size block
+      ::
+      =+  (rsh 3 2 -)
+      ::  Parse hex to atom
+      ::
+      `@ux`(rash - hex)
+    ::
+    ++  ip-port-to-cord
+      |=  [ip=@if port=@ud]
+      ^-  cord
+      =/  a=tape  (scow %if ip)
+      =/  b=json  (numb:enjs:format port)
+      ?>  ?=([%n *] b)
+      ?~  a  ''
+      (cat 3 (cat 3 (crip t.a) ':') p.b)
+    ::
+    ++  to-wall
+      |=  =tape
+      ^-  wall
```

Comment:
Huh, I'd expect this to be in the stdlib. Might be worth adding (in a separate PR), next to `+to-wain`.